PowerShell_Scripts/NameSurname_1more.ps1

```diff
@@ -0,0 +1,8 @@
+﻿Clear-Host
+$name = Read-Host 'What is your name?'
+$surName = Read-Host 'What is your surname?'
+Write-Host 'Indicate the file, where we need to save it'
+$path = Read-Host
+
+$getArray = Get-Content -Path $path -Raw -ErrorAction SilentlyContinue | ConvertFrom-Json 
```

Comment:
Generally speaking, _not_ failing on errors is a bad practice that leads to problems that are hard to detect. If a script enters an invalid state, it's almost always better to fail hard than to attempt to continue. For this reason `-ErrorAction SilentlyContinue`, while not being dangerous in this particular case, is rarely a good decision because it handles errors implicitly. 

Consider using `try...catch` to implement proper error handling. It can still allow the user to continue with a new file but it should prevent the situation where user has made a typo and didn't intend to create a new file.

Reply:
Thank you for the explanation. I've made a correction this way:
```suggestion
try {
    $getArray = Get-Content -Path $path -Raw | ConvertFrom-Json 
$NameSurname = @($name, $surName) + @($getArray) | ConvertTo-Json | Out-File $path <#C:\Users\Daria\Desktop\Tasks\PowerShell_Scripting\NameSurname.json#>
}
catch {

}
```
But I have a question: is it appropriate to leave the "catch" block empty?

Reply:
No. You should always handle an error unless you're absolutely sure that you _must not_ handle it, which is very rare.
The updated version uses the correct syntax but it still effectively hides an error, just like `-ErrorAction SilentlyContinue` does.

Try your hand at handling errors in the `catch` block. 😉